SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialConnectionManager.cpp

```diff
@@ -437,7 +442,11 @@ void USpatialConnectionManager::ConnectToLocator(FLocatorConfig* InLocatorConfig
 
 	ConfigureConnection ConnectionConfig(*InLocatorConfig, bConnectAsClient);
 
-	TSharedPtr<SpatialGDK::SpatialEventTracer> EventTracer = CreateEventTracer(InLocatorConfig->WorkerId);
+	TSharedPtr<SpatialGDK::SpatialEventTracer> EventTracer = nullptr;
+	if (USpatialGameInstance* GameInstance = GetTypedOuter<USpatialGameInstance>())
+	{
+		EventTracer = GameInstance->CreateEventTracer(ReceptionistConfig.WorkerId);
```

Comment:
Seems expensive? Can this be cached somewhere?

edit - seen similar comments above.